### PR TITLE
ci: fix and test with Wunused-macros

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6937

Describe changes:
- Remove unused macros as reported by compiler with `-Wunused-macros` and test in CI with default C flags

#10808 with green CI

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -324,6 +324,16 @@
         CFLAGS="$CFLAGS -std=gnu99"
     fi
 
+    # check if our compiler supports -Wunused-macros
+    AC_MSG_CHECKING(for -Wunused-macros support)
```
Comment (catenacyber) [inline configure.ac:328]: What do you think about this ?

Should we do it for other warning flags ?

File: src/runmode-af-xdp.c [hunk 2]
```diff
@@ -29,7 +29,6 @@
  * AF_XDP socket runmode
  *
  */
-#define PCAP_DONT_INCLUDE_PCAP_BPF_H 1
 #define SC_PCAP_DONT_INCLUDE_PCAP_H  1
```
Comment (catenacyber) [inline src/runmode-af-xdp.c:32]: Maybe this is not needed neither

File: configure.ac (hunk 1 above)
Comment (victorjulien) [reply configure.ac:328]: I don't think these belong in configure, other than, perhaps, as a `--enable-strict-compiler-flags` or similar. But generally we're not supposed to do too much with the CFLAGS. We probably already do too much right now.

File: src/runmode-af-xdp.c [hunk 3]
```diff
@@ -29,7 +29,6 @@
  * AF_XDP socket runmode
  *
  */
-#define PCAP_DONT_INCLUDE_PCAP_BPF_H 1
```
Comment (victorjulien) [inline src/runmode-af-xdp.c:32]: this is checked in pcap.h:
```
grep PCAP_DONT_INCLUDE_PCAP_BPF_H /usr/include/* -R
/usr/include/pcap/pcap.h:#ifndef PCAP_DONT_INCLUDE_PCAP_BPF_H
```

File: src/util-coredump-config.c [hunk 4]
```diff
@@ -24,7 +24,6 @@
  */
 
 #include "suricata-common.h"
-#define _FILE_OFFSET_BITS 64
```
Comment (victorjulien) [inline src/util-coredump-config.c:27]: this is used by system headers, but I'm not sure it's in the correct location here (or if it needed at all)

File: src/util-hash-lookup3.c [hunk 5]
```diff
@@ -64,8 +64,6 @@ on 1 byte), but shoehorning those bytes into integers efficiently is messy.
 # define HASH_BIG_ENDIAN 0
 #endif
 
-#define hashsize(n) ((uint32_t)1<<(n))
```
Comment (victorjulien) [inline src/util-hash-lookup3.c:67]: lets leave this file as close to vanilla as possible

File: src/app-layer-dnp3.c [hunk 6]
```diff
@@ -75,18 +72,6 @@ enum {
     DNP3_LINK_FC_UNCONFIRMED_USER_DATA
 };
 
-/* Reserved addresses. */
```
Comment (victorjulien) [inline src/app-layer-dnp3.c:78]: not sure about removing macros like this. @jasonish any thoughts?

File: src/util-coredump-config.c (hunk 4 above)
Comment (catenacyber) [reply src/util-coredump-config.c:27]: And CI is green...

File: src/util-hash-lookup3.c (hunk 5 above)
Comment (catenacyber) [reply src/util-hash-lookup3.c:67]: What do you mean by "close to vanilla" ?

File: src/runmode-af-xdp.c (hunk 3 above)
Comment (catenacyber) [reply src/runmode-af-xdp.c:32]: It does not hurt, but I do not think we need it here in runmode-af-xdp.c as it does not include <bpf.h>

But we need it in source-af-packet.c where both pcap.h and bpf.h are included

File: configure.ac (hunk 1 above)
Comment (catenacyber) [reply configure.ac:328]: Ok for something like `--enable-rust-strict`

But then, how do I get CI to get the most of it, as is done here ?

**Comment by catenacyber** [comment]
Continued in #10820

File: src/util-hash-lookup3.c (hunk 5 above)
Comment (victorjulien) [reply src/util-hash-lookup3.c:67]: this is a file we vendored in, so lets keep it as close to the original as possible

Comment (catenacyber) [reply src/util-hash-lookup3.c:67]: But this triggers rightfully the warning.

Is there some upstream to fix ?

**Pull request closed without merging**